src/s3contents.ts

```diff
@@ -1019,11 +994,12 @@ export class Drive implements Contents.IDrive {
         await this._s3Client.send(command);
 
       if (Contents) {
-        Contents.forEach(c => {
+        const promises = Contents.map(c => {
           const remainingFilePath = c.Key!.substring(path.length);
           // copy each file from old directory to new location
           this.copy_file(remainingFilePath, path, newFileName, this._name);
```

Comment:
Here, you need to `return` the `copy_file` promises in order to `await` them below.
```suggestion
          return this.copy_file(remainingFilePath, path, newFileName, this._name);
```